lib/shelley/src/Cardano/Wallet/Shelley/Api/Server.hs

```diff
@@ -395,3 +414,40 @@ server byron icarus shelley spl ntp =
                 pure NoContent
         getSettings'
             = Handler $ fmap ApiT $ liftIO $ getSettings spl
+
+postAnyAddress
+    :: NetworkId
+    -> ApiAddressData
+    -> Handler AnyAddress
+postAnyAddress net addrData = do
+    (addr, addrType) <- case addrData of
+        AddrEnterprise spendingCred ->
+            pure ( unAddress $
+                     CA.paymentAddress discriminant (spendingFrom spendingCred)
+                 , EnterpriseDelegating )
+        AddrRewardAccount stakingCred -> do
+            let (Right stakeAddr) =
+                    CA.stakeAddress discriminant (stakingFrom stakingCred)
+            pure ( unAddress stakeAddr, RewardAccount )
+        AddrBase spendingCred stakingCred ->
+            pure ( unAddress $ CA.delegationAddress discriminant
+                     (spendingFrom spendingCred) (stakingFrom stakingCred)
+                 , EnterpriseDelegating )
+    pure $ AnyAddress addr addrType (fromInteger netTag)
```

Comment:
Could we use the `NetworkId` type in `AnyAddress`?

Reply:
I had at the beginning networkId in AnyAddress and talked about this with @KtorZ  and there was idea to use networkId implicitely, ie., follow the network that the wallet instantiated with...